sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -547,6 +547,10 @@ valueExpression
     | left=valueExpression comparisonOperator right=valueExpression                          #comparison
     ;
 
+builtinOpExpression
+    : primaryExpression (CONCAT_PIPE primaryExpression)*                                       #concat
```

Comment:
Could you add the test cases to check whether we correctly follow the precedence like Oracle?

Ref: https://docs.oracle.com/cd/A87860_01/doc/server.817/a85397/operator.htm#1003584